kinto/core/resource/__init__.py

```diff
@@ -260,7 +260,7 @@ def collection_get(self):
         self._add_timestamp_header(self.request.response)
         self._add_cache_header(self.request.response)
         self._raise_304_if_not_modified()
-        self._raise_412_if_modified()
+        self._raise_412_if_modified(record={})  # ignore If-Match on non-existing
```

Comment:
I would adjust the comment, your explanation on the PR description was a lot better: for example, something like *collection is a resource that always exists"